state-reconstruct-fetcher/src/l1_fetcher.rs

```diff
@@ -243,81 +233,120 @@ impl L1Fetcher {
                 let mut latest_l2_block_number = U256::zero();
                 let mut previous_hash = None;
                 loop {
-                    // Break when reaching the `end_block` or on the receivement of a `ctrl_c` signal.
-                    if (disable_polling && current_l1_block_number > end_block_number)
-                        || cancellation_token.is_cancelled()
-                    {
+                    // Break on the receivement of a `ctrl_c` signal.
+                    if cancellation_token.is_cancelled() {
                         tracing::debug!("Shutting down main handler...");
                         break;
                     }
 
-                    // Create a filter showing only `BlockCommit`s from the [`ZK_SYNC_ADDR`].
-                    // TODO: Filter by executed blocks too.
-                    let filter = Filter::new()
-                        .address(ZK_SYNC_ADDR.parse::<Address>().unwrap())
-                        .topic0(event.signature())
-                        .from_block(current_l1_block_number)
-                        .to_block(current_l1_block_number + BLOCK_STEP);
+                    if let Some(end_block_number) = end_block {
+                        if current_l1_block_number > end_block_number {
```

Comment:
This whole thing is getting rather large, how about splitting and using some guard clauses instead?

Reply:
The scope thing itself might be a symptom of this nesting, so with some refactoring I think we can solve that. Re: guard clauses, There are a few ways to go about this, but roughly, I'm thinking something like:
```
if let None = end_block {
    ...
    continue
}

if current_l1_block_number > end_block_number {
    ...
    continue
}

// Do the main thing here (note: now it is not so heavily nested and thus more readable)
```

Reply:
well, the above won't work if the conditions have both if and else blocks... I tried to split the main loop (in [fc24d58514f343a5a4d56d7afbdcc761677f6570](https://github.com/eqlabs/zksync-state-reconstruct/pull/66/commits/fc24d58514f343a5a4d56d7afbdcc761677f6570)), but I doubt if it's all that readable, with the variables being modified across functions... Clippy also complains, about too many arguments - I suppose I could introduce a helper class... Thoughts?

Reply:
OK, next try: I can actually reorder the main loop blocks so that the largest one is last, and yes, with more continues it has a bit less nesting...